### PR TITLE
random reflection order

### PR DESCRIPTION
**Pull request by Jacyking**

### CODE REVIEW

File: main.cpp [hunk 1]
```diff
@@ -99,6 +99,26 @@ const char *ip = "127.0.0.1"; // your database ip
...
+TEST_CASE(flexible_reflection_order)
+{
+  dbng<mysql> mysql;
+  TEST_REQUIRE(mysql.connect(ip, "root", "12345", "testdb", /*timeout_seconds=*/5, 3306));
+  TEST_REQUIRE(mysql.create_datatable<jacy>());
+  TEST_REQUIRE(mysql.delete_records<jacy>());
+  TEST_REQUIRE(mysql.execute("alter table jacy modify name text first;"));
+  mysql.insert(jacy{ 666, "007" });
+  auto v = mysql.query<jacy>();
+  TEST_REQUIRE(v.size() > 0);
```
Comment (Jacyking) [inline main.cpp:117]: if we use "select *", it will return empty.

File: main.cpp [hunk 2]
```diff
@@ -99,6 +99,26 @@ const char *ip = "127.0.0.1"; // your database ip
...
+  TEST_REQUIRE(v.front().id == 666);
+  TEST_REQUIRE(v.front().name == "007");
+}
```
Comment (Jacyking) [inline main.cpp:120]: "insert" the same as "query", 
before: "insert into table values(?)"
after:"insert into table fields values(?)"
so, we don't have worry about whether the reflection order is the same as the database field order.

File: main.cpp [hunk 3]
```diff
@@ -99,6 +99,26 @@ const char *ip = "127.0.0.1"; // your database ip
...
+};
+REFLECTION(jacy, id, name);
+
+TEST_CASE(flexible_reflection_order)
+{
+  dbng<mysql> mysql;
+  TEST_REQUIRE(mysql.connect(ip, "root", "12345", "testdb", /*timeout_seconds=*/5, 3306));
+  TEST_REQUIRE(mysql.create_datatable<jacy>());
+  TEST_REQUIRE(mysql.delete_records<jacy>());
+  TEST_REQUIRE(mysql.execute("alter table jacy modify name text first;"));
```
Comment (Jacyking) [inline main.cpp:114]: Make the database order inconsistent with the reflection order.

File: main.cpp [hunk 4]
```diff
@@ -99,6 +99,27 @@ const char *ip = "127.0.0.1"; // your database ip
...
+  int id;
+  std::string name;
+};
+REFLECTION(test_order, name, id);
+
+TEST_CASE(random_reflection_order)
+{
+  dbng<mysql> mysql;
+  TEST_REQUIRE(mysql.connect(ip, "root", "12345", "testdb", /*timeout_seconds=*/5, 3306));
+  TEST_REQUIRE(mysql.execute("create table if not exists `test_order` (id int, name text);"));
```
Comment (Jacyking) [inline main.cpp:112]: reflection order "name id",
database order "id, name".
Make the database order inconsistent with the reflection order.

**Review by qicosmos — APPROVED** [review]
LGTM

**Pull request merged**